1_fetch/src/fetch_precip_nc.R

```diff
@@ -2,67 +2,60 @@
 #' @param view_polygon from the view_polygon target
 #' @param times the start and end time of the request
 fetch_precip_data <- function(ind_file, view_polygon, times) {
+  fabric <- webdata(
+    url = 'https://cida.usgs.gov/thredds/dodsC/stageiv_combined',
+    variables = "Total_precipitation_surface_1_Hour_Accumulation",
+    times = as.POSIXct(c(times$start,times$end), tz = 'UTC'))
 
   knife <- webprocess(algorithm = list('OPeNDAP Subset' = "gov.usgs.cida.gdp.wps.algorithm.FeatureCoverageOPeNDAPIntersectionAlgorithm"),
                       REQUIRE_FULL_COVERAGE = FALSE, wait = TRUE, OUTPUT_TYPE = 'netcdf')
 
   # create the stencil from the bbox of the view_polygon
   crs <- st_crs(view_polygon)
   bbox <- st_bbox(view_polygon)
-  as.stencil_pt <- function(box1, box2, col_name){
-    st_sf(st_sfc(st_point(c(bbox[[box1]], bbox[[box2]]), dim = "XY")), crs = crs) %>%
-      st_transform(crs = "+init=epsg:4326") %>%
-      st_coordinates() %>% as.vector() %>% data.frame(x = .) %>% setNames(col_name)
-  }
-  stencil_pts <- as.stencil_pt('xmin', 'ymax', 'up_left')
-  stencil_pts <- as.stencil_pt('xmax', 'ymax', 'up_right') %>% cbind(stencil_pts)
-  stencil_pts <- as.stencil_pt('xmax', 'ymin', 'low_right') %>% cbind(stencil_pts)
-  stencil_pts <- as.stencil_pt('xmin', 'ymin', 'low_left') %>% cbind(stencil_pts)
+  
+  precip_boundary <- "POLYGON ((-80.74672 19.80368, -119.023 23.117, -134.043 53.51192, -124.9736 55.90588, -114.5051 57.4236, -102.7265 57.82357, -91.28471 56.95466, -81.85043 55.19577, -73.63054 52.69872, -66.4331 49.53428, -59.95137 45.6187, -80.74672 19.80368))" %>%
```

Comment:
so this is unique to the Florence branch. Any reason we shouldn't have this in the main code?

Reply:
No. This is the boundary of the precip data source (stageiv). This will be the same for all.